Tests/KituraWebSocketTests/WebSocketClient.swift

```diff
@@ -18,90 +18,338 @@ import Foundation
 import NIO
 import NIOHTTP1
 import NIOWebSocket
+import Dispatch
+import XCTest
+@testable import KituraWebSocket
+
+#if os(Linux)
+    import Glibc
+#endif
 
 class WebSocketClient {
 
-    var httpClientHandler = HTTPClientHandler()
     let requestKey: String
     let host: String
     let port: Int
     let uri: String
     var channel: Channel? = nil
+    public var negotiateCompression: Bool
+    public var maxWindowBits: Int32
+    public var contextTakeover: ContextTakeover
+    public var maxFrameSize: Int
+
+    ///  This semaphore signals when the client successfully recieves the Connection upgrade response from remote server
```

Comment:
We should not have documentation comments (starting with `///`, and not `//`) here.